src/main.rs

```diff
@@ -80,7 +90,53 @@ fn main() {
         Rank::InfoPlusProbability => roget::Rank::InfoPlusProbability,
         Rank::ExpectedInformation => roget::Rank::ExpectedInformation,
     };
-    play(move || solver.build(), args.games);
+    if args.interactive {
+        play_interactive(move || solver.build());
+    } else {
+        play(move || solver.build(), args.games);
+    }
+}
+
+fn play_interactive<G>(mk: impl FnOnce() -> G)
+where
+    G: Guesser,
+{
+    let mut guesser = mk();
```

Comment:
Since we only instantiate one game this way, there's no need for a "maker" (`impl FnOnce() -> G`). Instead, just take a `G` directly, so this becomes:

```suggestion
fn play_interactive(guesser: impl Guesser) {
```